molecule/driver/hetznercloud.py

```diff
@@ -0,0 +1,190 @@
+        newly created instance because you must pre-register that key with your
+        Hetzner Cloud account access.
+
+    .. code-block:: bash
+
+        $ pip install 'molecule[hetznercloud]'
+
+    Change the options passed to the ssh client.
+
+    .. code-block:: yaml
+
+        driver:
+          name: hetznercloud
+          ssh_connection_options:
+            -o ControlPath=~/.ansible/cp/%r@%h-%p
+
+    .. important::
+
+        Molecule does not merge lists, when overriding the developer must
+        provide all options.
+
+    Provide the files Molecule will preserve upon each subcommand execution.
+
+    .. code-block:: yaml
+
+        driver:
+          name: hetznercloud
+          safe_files:
+            - foo
+    """  # noqa
+
+    def __init__(self, config):
+        super(HetznerCloud, self).__init__(config)
+        self._name = 'hetznercloud'
+
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def login_cmd_template(self):
+        connection_options = ' '.join(self.ssh_connection_options)
+
+        return ('ssh {{address}} '
+                '-l {{user}} '
+                '-p {{port}} '
+                '{}').format(connection_options)
+
+    @property
+    def default_safe_files(self):
+        return [
+            self.instance_config,
+        ]
+
+    def _get_ssh_connection_options(self):
+        return [
+            '-o UserKnownHostsFile=/dev/null',
+            '-o ControlMaster=auto',
+            '-o ControlPersist=60s',
+            '-o IdentitiesOnly=no',  # rely on ssh-agent instead
+            '-o StrictHostKeyChecking=no',
+        ]
+
+    @property
+    def default_ssh_connection_options(self):
+        return self._get_ssh_connection_options()
+
+    def login_options(self, instance_name):
+        d = {'instance': instance_name}
+
+        return util.merge_dicts(d, self._get_instance_config(instance_name))
+
+    def ansible_connection_options(self, instance_name):
+        try:
+            d = self._get_instance_config(instance_name)
+
+            return {
+                'ansible_user': d['user'],
+                'ansible_host': d['address'],
+                'ansible_port': d['port'],
+                'connection': 'ssh',
+                'ansible_ssh_common_args':
+                ' '.join(self.ssh_connection_options),
+            }
+        except StopIteration:
+            return {}
+        except IOError:
+            # Instance has yet to be provisioned , therefore the
+            # instance_config is not on disk.
+            return {}
+
+    def _get_instance_config(self, instance_name):
+        instance_config_dict = util.safe_load_file(
+            self._config.driver.instance_config)
+
+        return next(item for item in instance_config_dict
+                    if item['instance'] == instance_name)
```

Comment:
You probably want to have a default value to prevent `StopIteration`:
```suggestion
                    if item['instance'] == instance_name, ())
```

Reply:
Unrelated since this is a copy/pasta and all other files do that. I'm not refactoring here.